src/psbtv2.ts

```diff
@@ -982,8 +983,7 @@ export class PsbtV2 extends PsbtV2Maps {
       }
     }
 
-    this.inputMaps.push(map);
-    this.PSBT_GLOBAL_INPUT_COUNT = this.inputMaps.length;
+    this.PSBT_GLOBAL_INPUT_COUNT = this.inputMaps.push(map);
```

Comment:
clever! the fact that `push` returns the length usually is the cause of bugs for me because I forget about it. Nice to see it used deliberately.